ts/packages/node/src/stores/LaunchDarkly.ts

```diff
@@ -0,0 +1,173 @@
+import { Config, ConfigStore, ConfigStoreQuery } from '@configu/ts';
+import axios, { Axios } from 'axios';
+import assert from 'node:assert';
+import * as fs from 'fs';
+
+export type LaunchDarklyConfigStoreConfigurations = { apitoken: string; defaultproject: string } | string;
+
+export class LaunchDarklyConfigStore extends ConfigStore {
+  private client: Axios;
+  private readonly projectKey: string;
+  private environments: string[];
+
+  constructor(configurations: LaunchDarklyConfigStoreConfigurations) {
+    super('launch-darkly');
+    const configs =
+      typeof configurations === 'string'
+        ? JSON.parse(fs.readFileSync('ldc.json').toString())[configurations]
+        : configurations;
+    this.projectKey = configs.defaultproject;
+    this.client = axios.create({
+      baseURL: `${configs.server ? configs.server : 'https://app.launchdarkly.com'}/api/v2`,
+      headers: {
+        Authorization: configs.apitoken,
+      },
+      responseType: 'json',
+    });
+    this.environments = [];
+  }
+
+  private async patchUpdate(config: Config, patchData: Record<string, any>) {
+    const { data } = await this.client.patch(`/flags/${this.projectKey}/${config.key}`, patchData, {
+      headers: { 'Content-Type': 'application/json; domain-model=launchdarkly.semanticpatch' },
+    });
+    return data;
+  }
+
+  private async updateEnvironments() {
+    const { data: environments } = await this.client.get(`/projects/${this.projectKey}/environments`);
+    this.environments = [...this.environments, ...environments.items.map((env: any) => env.key)];
+  }
+
+  private async getFeatureFlag(config: Config) {
+    const { data: featureFlag } = await this.client.get(`/flags/${this.projectKey}/${config.key}`);
+    return featureFlag;
+  }
+
+  private async createEnvironment(config: Config) {
+    await this.client.post(`/projects/${this.projectKey}/environments`, {
+      color: 'FFFFFF',
+      key: config.set,
+      name: config.set,
+    });
+    this.environments = [...this.environments, config.set];
+  }
+
+  private async createFeatureFlag(config: Config) {
+    const createData = {
+      clientSideAvailability: {
+        usingEnvironmentId: true,
+        usingMobileKey: true,
+      },
+      key: config.key,
+      name: config.key,
+      variations: [{ value: config.value }, { value: '' }],
```

Comment:
The empty string variant is very arbitrary, I'm not sure if this is the right approach. If we do this then we can never create a boolean or any other type of feature flag by using configu:
![image](https://github.com/configu/configu/assets/47181182/91f45e8b-568e-4c9f-b89d-def38b71207a)

We need to consider if this makes sense for whoever uses this store.

Reply:
We always use strings in their DB and will take care of the type from our end via CFGU